### DISCUSSION TITLE
Return alt text when selecting asset in `Craft.createElementSelectorModal`

### DISCUSSION THREAD

**Discussion by leevigraham** [craftcms/cms#12600] [Ideas] [answered]
Currently when selecting an asset in the `Craft.createElementSelectorModal` the returned object is:

![image](https://user-images.githubusercontent.com/25124/215929082-1d87e2d6-37a2-4fe4-a784-06c32a1ad91f.png)

Given `alt` is a native field it should be included in the returned data. The `alt` text can then be used in WYSIWYG editor implementations for better accessibility.

Eg: ckEditor:

![image](https://user-images.githubusercontent.com/25124/215929179-ce332c16-55ee-46e7-a103-6867bc907233.png)

**Comment by brandonkelly** [comment, marked as answer]
Individual element types don’t have control over which data gets returned there. However I’ve added a `data-alt` attribute to asset element components for the next release, which you could access via `element.$element.data('alt')`. (f59da952885dd90da02887c1bce2e6765ce67926)

**Reply by leevigraham** [reply]
Is there a way to get the height and width as well?

**Reply by brandonkelly** [reply]
Already there (`data-image-width` and `data-image-height`).

**Reply by leevigraham** [reply]
Perfect… I'm going to extend my `CraftImage` CKEditor plugin with these values which will hopefully address: https://github.com/ckeditor/ckeditor5/issues/5154
